pkg/apis/core/v1alpha1/federatedcluster_types.go

```diff
@@ -43,7 +43,13 @@ type FederatedClusterSpec struct {
 	//
 	// This can be left empty if the cluster allows insecure access.
 	// +optional
-	SecretRef *apiv1.LocalObjectReference `json:"secretRef,omitempty"`
+	SecretRef *LocalSecretReference `json:"secretRef,omitempty"`
+}
+
+// LocalSecretReference contains information to reference secret in the
+// cluster registry.
```

Comment:
It seems not correct in comments. It is used to refer a `secret` instead of `registry`

Reply:
@xunpan my bad. Let me change `registry` to `secret`